apinetlet/controllers/network_controller_test.go

```diff
@@ -67,8 +73,12 @@ var _ = Describe("NetworkController", func() {
 		Expect(k8sClient.Status().Update(ctx, apiNetNetwork)).To(Succeed())
 
 		By("waiting for the network to reflect the allocated vni")
-		Eventually(Object(network)).
-			Should(HaveField("ObjectMeta.Annotations", HaveKeyWithValue(onmetalapinetv1alpha1.OnmetalAPINetworkVNIAnnotation, strconv.FormatInt(int64(vni), 10))))
+
+		Eventually(func(g Gomega) {
+			g.Expect(Get(network)).Should(Succeed())
+			g.Expect(network.Status.State).Should(BeEquivalentTo(networkingv1alpha1.NetworkStateAvailable))
+			g.Expect(network.Spec.ProviderID).Should(BeEquivalentTo(strconv.FormatInt(int64(vni), 10)))
+		})
```

Comment:
We should keep the `Eventually` / `Should` `HaveField` approach as
* It's in line with the controller-runtime tests using `komega`'s `Object`
* Is more declarative
* Produces more output in case multiple conditions fail.

Additionally, in your current `Eventually` a `.Should(Succeed())` is missing, resulting in the assertion never actually being tested.

To have multiple assertions within `Should`, use `SatisfyAll` which accepts multiple matchers.